serverless/src/submitSwodlrOrder/handler.js

```diff
@@ -0,0 +1,190 @@
+  generateL2RasterProduct(cycle: $cycle, pass: $pass, scene: $scene, outputGranuleExtentFlag: $outputGranuleExtentFlag, outputSamplingGridType: $outputSamplingGridType, rasterResolution: $rasterResolution, utmZoneAdjust: $utmZoneAdjust, mgrsBandAdjust: $mgrsBandAdjust) {
+    cycle
+    pass
+    scene
+    outputGranuleExtentFlag
+    outputSamplingGridType
+    rasterResolution
+    utmZoneAdjust
+    mgrsBandAdjust
+    id
+    timestamp
+    status {
+        id
+        timestamp
+        state
+        reason
+    }
+  }
+}`
+
+/**
+ * Submits an order to SWODLR
+ * npm run invoke-local -- --function submitSwodlrOrder -p tmp/test_swodlr_order.json
+ * @param {Object} event Queue messages from SQS
+ * @param {Object} context Methods and properties that provide information about the invocation, function, and execution environment
+ */
+const submitSwodlrOrder = async (event, context) => {
+  // https://stackoverflow.com/questions/49347210/why-aws-lambda-keeps-timing-out-when-using-knex-js
+  // eslint-disable-next-line no-param-reassign
+  context.callbackWaitsForEmptyEventLoop = false
+
+  // Retrieve a connection to the database
+  const dbConnection = await getDbConnection()
+
+  const { Records: sqsRecords = [] } = event
+
+  if (sqsRecords.length === 0) return
+
+  console.log(`Processing ${sqsRecords.length} order(s)`)
+
+  await sqsRecords.forEachAsync(async (sqsRecord) => {
+    const { body } = sqsRecord
+
+    // Destruct the payload from SQS
+    const {
+      accessToken,
+      id
+    } = JSON.parse(body)
+
+    try {
+      // Fetch the retrieval id that the order belongs to so that we can provide a link to the status page
+      const retrievalRecord = await dbConnection('retrieval_orders')
+        .first(
+          'retrievals.id',
+          'retrievals.environment',
+          'retrieval_collections.access_method',
+          'retrieval_collections.collection_id',
+          'retrieval_orders.granule_params'
+        )
+        .join('retrieval_collections', { 'retrieval_orders.retrieval_collection_id': 'retrieval_collections.id' })
+        .join('retrievals', { 'retrieval_collections.retrieval_id': 'retrievals.id' })
+        .where({
+          'retrieval_orders.id': id
+        })
+
+      const {
+        access_method: accessMethod,
+        collection_id: collectionConceptId,
+        environment: earthdataEnvironment,
+        granule_params: granuleParams
+      } = retrievalRecord
+
+      const { type, url: swodlrUrl } = accessMethod
+
+      // No need for Accessmethod at this stage of dev.
+      const granuleInfo = await constructOrderPayload({
+        collectionConceptId,
+        granuleParams,
+        accessToken,
+        earthdataEnvironment
+      })
+
+      const { orderItems } = granuleInfo
+
+      await orderItems.forEachAsync(async (granule) => {
+        const { granuleUr, granuleConceptId } = granule
+
+        const splitTitle = granuleUr.split('_')
```

Comment:
I really wanna talk to the swodlr team to make sure this is absolutely the only way its super odd that we have to do this to get this information out

Reply:
It'll probably be a week until we can get a meeting set up.